core/src/main/scala/kafka/server/BrokerMetadataCheckpoint.scala

```diff
@@ -24,7 +24,7 @@ import java.util.Properties
 import kafka.utils._
 import org.apache.kafka.common.utils.Utils
 
-case class BrokerMetadata(brokerId: Int)
+case class BrokerMetadata(brokerId: Int, clusterId: String)
```

Comment:
Could we make this an `Option` since it is not guaranteed to be present?

Reply:
@hachikuji I initially wanted to use an Option here but did not eventually. The reasoning behind is that the `clusterId` won't be present only once (first start after the upgrade) so I though that using an Option is a bit overkill here.

That being said, I am fine with adding it.

Reply:
Done.

Reply:
If we could figure out how to remove the one nullable case, that would be another option. I just think these assumptions tend to be a bit brittle over time, so I prefer to use explicit types.